### ISSUE TITLE
Unable to continuously obtain coordinates from this.$watchLocation(opts)

### ISSUE THREAD

**Issue by chuabingquan** [scaccogatto/vue-geolocation#15]
Hello,

Is it possible to continuously obtain new coordinate values from this.$watchLocation(opts)?
The current design using promises seems to only return a value at first invocation, then does not supply any new coordinates thereafter.

**Comment by scaccogatto** [comment]
You are right, the watch feature was introduced by @hkd987 and it is not what it should be. We could use reactive data and provide a real watch experience for this plugin. PR are welcome!

**Comment by hkd987** [comment]
How are you calling the method? From what part of vue? 

I’d like to get more details maybe I can come up with a better solution. 

When I tested this I was only able to get the watch position to work from a cell phone. When used on a desktop it will only work when you open a new tab. 

https://stackoverflow.com/questions/24019286/navigator-geolocation-watchposition-frequency

Have you tried putting the call inside a computed property?

**Comment by chuabingquan** [comment]
@hkd987 I called the method in a component's mounted lifecycle hook and it returned a promise that resolves only the first value returned by the native watch location API.

Nope, I did not put it within a computed property, but does it make a difference as the native watch location API is wrapped in a promise that resolves only once?

But as @scaccogatto have mentioned, maybe we could use reactive data to provide a real watch experience for this plugin? 😄
